go/vt/worker/split_diff.go

```diff
@@ -398,7 +402,7 @@ func (sdw *SplitDiffWorker) diff(ctx context.Context) error {
 		sdw.destinationSchemaDefinition, err = sdw.wr.GetSchema(
 			shortCtx, sdw.destinationAlias, nil /* tables */, sdw.excludeTables, false /* includeViews */)
 		cancel()
-		rec.RecordError(err)
+		sdw.markAsFailed(rec, err)
```

Comment:
Here you're not handling the "err == nil" case: If err == nil, then rec.RecordError() returned early and nothing happened.

There are similar comments below.

In practice what probably happens here is:
- your new code runs and sets the state to "diff is about to fail"
- diff actually finishes all the way to the end and then state is set to success

Ideally, the state transition code shouldn't allow that in the first place. But so far it's just doing simple overrides and no fancy checks.

To keep things simple, I suggest the following: Call your new method here only if err != nil.